src/Serilog.Sinks.AzureDataExplorer.Tests/AzureDataExplorerDurableSinkE2ETests.cs

```diff
@@ -0,0 +1,265 @@
+            log.Error(identifier + "Exception occurred", "Zohar Processed {@Position} in {Elapsed:000} ms.", position,
+                elapsedMs);
+            log.Debug(identifier + "Processed {@Position} in {Elapsed:000} ms. ", position, elapsedMs);
+            log.Verbose(identifier + "Processed {@Position} in {Elapsed:000} ms.", position, elapsedMs);
+            log.Information(identifier + "Processed {@Position} in {Elapsed:000} ms.", position, elapsedMs);
+            log.Warning(identifier + "Processed {@Position} in {Elapsed:000} ms.", position, elapsedMs);
+            log.Error(identifier + "Exception occurred", "Zohar Processed {@Position} in {Elapsed:000} ms.", position,
+                elapsedMs);
+            log.Debug(identifier + "Processed {@Position} in {Elapsed:000} ms. ", position, elapsedMs);
+            log.Verbose(identifier + "Processed {@Position} in {Elapsed:000} ms.", position, elapsedMs);
+            log.Information(identifier + "Processed {@Position} in {Elapsed:000} ms.", position, elapsedMs);
+        }
+
+        await Task.Delay(10000);
+
+        int lineCount = 0;
+        foreach (string file in Directory.EnumerateFiles(Directory.GetCurrentDirectory() + "/" + randomInt, "*.clef"))
+        {
+            lineCount += System.IO.File.ReadLines(file).Count();
+        }
+        Assert.Equal(2, lineCount);
+
+        int noOfRecordsIngested =
+            GetNoOfRecordsIngestedInAdx(identifier);
+        Assert.Equal(2, noOfRecordsIngested);
+    }
+
+    private int GetNoOfRecordsIngestedInAdx(string searchString)
+    {
+        var kcsbEx = new KustoConnectionStringBuilder(
+                AzureDataExplorerSinkOptionsExtensions.GetClusterUrl(
+                    Environment.GetEnvironmentVariable("ingestionURI")),
+                Environment.GetEnvironmentVariable("databaseName"))
+            .WithAadApplicationKeyAuthentication(Environment.GetEnvironmentVariable("appId"),
+                Environment.GetEnvironmentVariable("appKey"), Environment.GetEnvironmentVariable("tenant"));
+        var noOfRecordsIngested = 0;
+        using (var queryProvider = KustoClientFactory.CreateCslQueryProvider(kcsbEx))
+        {
+            var query = m_generatedTableName + " | where Message contains '" + searchString + "' | count";
+            var clientRequestProperties = new ClientRequestProperties() { ClientRequestId = Guid.NewGuid().ToString() };
+            using (var reader = queryProvider.ExecuteQuery(Environment.GetEnvironmentVariable("databaseName"), query,
+                       clientRequestProperties))
+            {
+                // Read HowManyRecords
+                while (reader.Read())
+                {
+                    noOfRecordsIngested = (int)reader.GetInt64(0);
+                }
+            }
+        }
+
+        return noOfRecordsIngested;
+    }
+
+    public void Dispose()
+    {
+        var kcsb = new KustoConnectionStringBuilder(
```

Comment:
A dispose is a cleanup method , would it make sense to have an Admin client in KCSB in a Before/After or in the constructor and use it here than instantiating one here ?

Reply:
make sense. done the change.